src/Regridder.jl

```diff
@@ -0,0 +1,388 @@
+# Code from ClimaCoupler Regridder.jl
```

Comment:
Given that a lot of this is copied from ClimaCoupler, it would be preferable to move this to ClimaCore to avoid duplication. @simonbyrne, what do you think?

Reply:
@simonbyrne @juliasloan25 perhaps we can merge this PR, and then a later PR would move the duplicate code from the coupler and climalsm into climacore?